protoc-gen-go/generator/generator.go

```diff
@@ -847,11 +852,27 @@ func (g *Generator) WrapTypes() {
 		if fd == nil {
 			g.Fail("could not find file named", fileName)
 		}
-		fd.index = len(g.genFiles)
+		fingerprint, err := fingerprintProto(fd.FileDescriptorProto)
+		if err != nil {
+			g.Error(err)
+		}
+		fd.fingerprint = fingerprint
 		g.genFiles = append(g.genFiles, fd)
 	}
 }
 
+// fingerprintProto returns a fingerprint for a message.
+// The fingerprint is intended to prevent conflicts between generated fileds,
+// not to provide cryptographic security.
+func fingerprintProto(m proto.Message) (string, error) {
+	b, err := proto.Marshal(m)
```

Comment:
Shouldn't we be setting the deterministic flag? 

Note that this can still be non-deterministic if the `FileDescriptorProto` as given to us has non-deterministic serialization in an unknown field or field extension. However, this is very unlikely to happen.

Reply:
Also, if it is non-deterministic, it just means we create two variables for what are semantically the same proto, right? No a bad failure scenario.

Reply:
If it's non-deterministic, we'll generate a different var name each time we compile the source file.  Not really the end of the world.

Reply:
Thinking about it more, doesn't that break assumptions about how Bazel works? It relies on a generation step to produce deterministic output, right?

Reply:
As for setting the deterministic flag...we could, but we're already marshalling this exact proto elsewhere without it, and I'd rather make that change separately if we're going to do it.

Reply:
It seems we're not any more or less deterministic than before. SGTM.